### PR TITLE
Delete event

### LINKED ISSUE #1: D for Delete
In the existing key event handling code in `src/ui.rs`, in the `event_action` function, add a match clause for `D` key and add a Delete action to the Action enum and implement a `delete_current_image` function in `src/program.rs. `delete_current_image` should be a method on Program, taking a mutable reference to self.

The method will need to delete the file, remove the path from the `self.images` path vector, update the `self.index` and call render. Pretty similar to the current `move_image` function. Remember to handle the case where we end up with a Vec length of zero or one, and also that special case that we're deleting the last path in the Vector. 

An error in deleting the file should probably not cause the program to exit, so we probably don't want to use the `?` operator in error handling, but perhaps instead print an error message to terminal use the `eprintln!` macro.

### PR DESCRIPTION
**Pull request by gurgalex**
Deletes the current image on keypress D.

For now, it prints out an error message about not being able to delete an image if there are no images to delete.

Fixes #1

### CODE REVIEW

File: src/program.rs [hunk 1]
```diff
@@ -144,6 +145,43 @@ impl Program {
...
+        // Check if there is an image to delete
+        if self.images.is_empty() {
+            return Err("no images to delete".to_string());
+        }
+
+        // Retrieve current image
+        assert!(self.index < self.images.len());
+        let current_imagepath = self.images.get(self.index).unwrap_or_else(|| {
+            panic!(format!(
+                "image index {} < max image index {}",
```
Comment (Davejkane) [inline src/program.rs]: Should this `<` be the other way around? `>`

File: src/program.rs [hunk 2]
```diff
@@ -144,6 +145,43 @@ impl Program {
...
+        // If we've reached past here, there was no error deleted the image
+
+        // Only if successful, remove image from tracked images
+        self.images.remove(self.index);
+
+        // Adjust our view
+        self.decrement(1)?;
+
+        // Image successfully deleted, we're done
+        Ok(())
```
Comment (Davejkane) [inline src/program.rs]: If you leave the last line as `self.decrement(1)`, without the `?;` then we don't need this superfluous `Ok(())`

File: src/ui.rs [hunk 3]
```diff
@@ -95,6 +97,10 @@ pub fn event_action(state: &mut State, event: &Event) -> Action {
             keycode: Some(Keycode::M),
             ..
         } => Action::Move,
+        Event::KeyDown {
+            keycode: Some(Keycode::D),
+            ..
+        } => Action::Delete,
```
Comment (Davejkane) [inline src/ui.rs:107]: I know I didn't mention it in the spec, but can we match on the Delete key as well please?

**Review by Davejkane — CHANGES_REQUESTED** [review]
Awesome, thanks so much for getting involved. Really nice code, something I can learn from. I've requested a couple of changes, very minor stuff.

File: src/ui.rs (hunk 3 above)
Comment (gurgalex) [reply src/ui.rs:107]: Thanks for the spec!
It's great info on what needs done.

Should Backspace be added for delete as well?

Comment (Davejkane) [reply src/ui.rs:107]: 🤔 Not sure. What do you think? I'm leaning towards no.

**Review by Davejkane — APPROVED** [review]
LGTM. I think we should just leave backspace out for now.

File: src/ui.rs (hunk 3 above)
Comment (gurgalex) [reply src/ui.rs:107]: http://cdn.osxdaily.com/wp-content/uploads/2012/02/mac-keyboard.jpg

Hmm, wonder what the delete keycode is on an Apple keyboard.

PC keyboards have a Delete and backspace key.

Not an issue right now though.

**Pull request merged**